jujugui/static/gui/src/app/state/state.js

```diff
@@ -252,10 +260,20 @@ const State = class State {
           return false;
         }
       }
-      return found;
+      return {
+        dispatchers: found,
+        key: key
+      };
     }
     // Recurse up the dispatcher tree to find matching dispatchers.
-    const dispatchers = findDispatchers(key, this._dispatchers);
+    const matchingDispatchers = findDispatchers(key, this._dispatchers);
+    // If this dispatcher has already been called for this state then don't call
+    // it again. e.g. 'gui.inspector' should only be called once for
+    // 'gui.inspector.id' and 'gui.inspector.unit'.
+    if (dispatched.indexOf(matchingDispatchers.key) > -1) {
```

Comment:
Why this check? Could you not rely on the the `if (!dispatchers) {` check below?